src/components/PostData/PostInfo.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useContext } from 'react';
+import classNames from 'classnames';
+import { Post } from '../../types/Post';
+import {
+  ActionTypes,
+  DispatchContext,
+  StateContext,
+} from '../../reducer/store';
+
+type Props = {
+  post: Post
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const dispatch = useContext(DispatchContext);
+  const { selectedPost } = useContext(StateContext);
+  const isPostChosen = selectedPost?.id === post.id;
+
+  const onClickHandler = useCallback(() => {
```

Comment:
The dependency array for useCallback is missing the 'dispatch' dependency. It's recommended to include all the function's dependencies to ensure that it behaves correctly.